module/signature/aggregation.go

```diff
@@ -105,6 +107,7 @@ func (s *SignatureAggregatorSameMessage) VerifyAndAdd(signer int, sig crypto.Sig
 	// signature is new
 	ok, err := s.publicKeys[signer].Verify(sig, s.message, s.hasher)
 	if ok {
+		s.cashedSignature = nil
 		s.indexToSignature[signer] = string(sig)
```

Comment:
I would really prefer to move 
```
s.cashedSignature = nil
s.indexToSignature[signer] = string(sig)
```
into function so invalidation of those data structures happens in atomic way.